src/kiwixapp.cpp

```diff
@@ -105,10 +105,14 @@ void KiwixApp::init()
         m_library.asyncLoadMonitorDir(monitorDir);
     });
     QString monitorDir = m_settingsManager.getMonitorDir();
-    if (monitorDir != "") {
-        m_library.setMonitorDirZims(m_library.getLibraryZimsFromDir(monitorDir));
-        m_watcher.addPath(monitorDir);
-        m_library.asyncLoadMonitorDir(monitorDir);
+    QString downloadDir = m_settingsManager.getDownloadDir();
+    auto dirList = QSet<QString>({monitorDir, downloadDir});
+    for (auto dir : dirList) {
+        if (dir != "") {
+            m_library.setMonitorDirZims(dir, m_library.getLibraryZimsFromDir(dir));
+            m_watcher.addPath(dir);
+            m_library.asyncLoadMonitorDir(dir);
+        }
```

Comment:
I'm on the assumption that adding the same book using manipulator will not add it twice here. Is it possible to duplicate?

Reply:
I think we should be mostly ok:
- QFileSystemWatcher handles duplicate correctly
- `libkiwix::Library` handles duplicate (`Manager` doesn't but I think it is ok)
- `Library::m_monitorDirZims` splits different directory in different list (and handles duplicate also as it is now a QMap)

However, the problem is race condition. If we have two `loadMonitorDir` running in the same time, `Library::m_monitorDirZims` is not protected against multi threaded access. (`libkiwix::Library` is).
And we launch at least two `loadMonitorDir` in the same time here with the two `m_library.asyncLoadMonitorDir(dir);`
And it would be even more tricky if `monitorDir` == `downloadDir`.
Properly protecting `loadMonitorDir` from concurrent executions **should** be enough.